### PR TITLE
Updating schemas to WD06 and minor bug fixes

### LINKED ISSUE #91: Update schemas to 2.1 WD06
* empty dictionaries are now prohibited just like empty lists

* socket-ext.options now has some requirements around its keys/values

* x509-certificate objects MUST have at least 1 object specific property (non-common, non-custom) other than 'type'

### PR DESCRIPTION
**Pull request by zrush-mitre**
Closes #91

### CODE REVIEW

**Comment by zrush-mitre** [comment]
Github shows some of my changes as odd indentation, though the files in my fork don't have those indents. I'm not sure if it's a github bug, or something else, but I have no idea how to fix it.

File: schemas/sdos/observed-data.json [hunk 1]
```diff
@@ -36,9 +36,8 @@
           "maximum": 999999999
         },
         "objects": {
-          "type": "object",
-          "description": "A dictionary of Cyber Observable Objects that describes the single 'fact' that was observed.",
-          "minProperties": 1,
+          "allOf": [{ "$ref": "../common/dictionary.json" }],
+	  "description": "A dictionary of Cyber Observable Objects that describes the single 'fact' that was observed.",
```
Comment (clenk) [inline schemas/sdos/observed-data.json]: There's still some weird indenting on this line.

File: schemas/common/cyber-observable-core.json [hunk 2]
```diff
@@ -54,7 +54,7 @@
     }
   },
   "patternProperties": {
-    "^[a-z0-9_]{0,246}_bin$": {
```
[the remark this answers is not in the document]
Comment (zrush-mitre) [reply schemas/common/cyber-observable-core.json]: On page 37 of WD06, it specifies "All type names, property names, object names, and vocabulary terms MUST be between three and 250 characters long." I figured that included basically any string being used in a STIX schema.

[the remark this answers is not in the document]
Comment (clenk) [reply schemas/common/cyber-observable-core.json]: Right, but the `_bin` will push it over. With `"^[a-z0-9_]{3,250}_bin$"` you could have a 254 character long value.

[the remark this answers is not in the document]
Comment (zrush-mitre) [reply schemas/common/cyber-observable-core.json]: Ah that makes sense. Reverting changes.

**Comment by clenk** [comment]
Thanks, @zrush-mitre!

**Pull request merged**